.github/workflows/sonarcube.yaml

```
@@ -0,0 +1,37 @@
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle-
+
+      - name: Build and Test
+        run: ./gradlew build jacocoTestReport --info
+
+      - name: SonarQube Analysis
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew sonar --info
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)